pkg/controller/provider-network.go

```diff
@@ -4,126 +4,58 @@ import (
 	"context"
 	"fmt"
 
-	corev1 "k8s.io/api/core/v1"
-	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"github.com/kubeovn/kube-ovn/pkg/util"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
-
-	kubeovnv1 "github.com/kubeovn/kube-ovn/pkg/apis/kubeovn/v1"
-	"github.com/kubeovn/kube-ovn/pkg/util"
 )
 
-func (c *Controller) enqueueUpdateProviderNetwork(_, obj interface{}) {
-	if !c.isLeader() {
-		return
-	}
-
-	key, err := cache.MetaNamespaceKeyFunc(obj)
+func (c *Controller) resyncProviderNetworkStatus() {
+	klog.Infof("start to sync ProviderNetwork status")
+	pns, err := c.providerNetworksLister.List(labels.Everything())
```

Comment:
Can we refactor this function and make provider-network status updates triggered by node label updates?

Reply:
In my opinion, it is not necessary, because the state is only the presentation of the final result, and the delay in the presentation of the result will not cause functional impact. We should minimize the update operation to avoid the disturbance to the cluster, especially in the case of a large cluster size.

Reply:
> Can we refactor this function and make provider-network status updates triggered by node label updates?

@zhangzujian In my opinion, it is not necessary, because the state is only the presentation of the final result, and the delay in the presentation of the result will not cause functional impact. We should minimize the update operation to avoid the disturbance to the cluster, especially in the case of a large cluster size.